ring/util_test.go

```diff
@@ -424,3 +425,32 @@ func TestGetTokenDistance(t *testing.T) {
 		require.Equal(t, testData.expected, distance)
 	}
 }
+
+func TestSearchToken(t *testing.T) {
```

Comment:
Have you verified that this behavior matches the old implementation?

Reply:
I did, but just to triple-check myself since this is a core part of the code, https://github.com/grafana/dskit/pull/408/commits/9cec3d0c18161df552091809b7e31ad1cea7c96e rolls back to the old function, and adds a test that compares the results of the new and old functions over 1M random keys. All tests passed [here](https://drone.grafana.net/grafana/dskit/3289/1/4). https://github.com/grafana/dskit/pull/408/commits/1275009a299573b0bfc3a88f97d5aeb016e2fd78 reverts those changes.